### PR TITLE
fix: fix return types for JsonSerializable::jsonSerialize methods

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
FIxing errors related to the wrong implementation for methods that implement `JsonSerializable::jsonSerialize` method.

### CODE REVIEW

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by poyuki — APPROVED** [review]
LGTM

**Review by poyuki — COMMENTED** [review]
whoops tests looks failed, could you please also validate what happens?

File: models/classes/event/UserUpdatedEvent.php [hunk 1]
```diff
@@ -27,15 +29,9 @@
 
 class UserUpdatedEvent implements Event, JsonSerializable
 {
+    protected string $user;
```
Comment (poyuki) [inline models/classes/event/UserUpdatedEvent.php:32]: constructor expects `core_kernel_classes_Resource` as user

Comment (poyuki) [reply models/classes/event/UserUpdatedEvent.php:32]: also on raw https://github.com/oat-sa/tao-core/pull/3714/files/ae7d396a923fdcc305c4d73884ad9533caf42be4#diff-352e46b6d74a923229c4a21ca52b7900f6c80aa05a26e5affebbe665ccbfad09R61 we expecting not a string

File: models/classes/security/xsrf/Token.php [hunk 2]
```diff
@@ -43,49 +56,39 @@ public function __construct($data = [])
         }
     }
 
-    /**
-     * Set the value of the token.
-     *
-     * @param string $token
-     */
-    public function setValue($token)
+    public function setValue(string $token): void
```
Comment (Karol-Stelmaczonek) [inline models/classes/security/xsrf/Token.php:59]: Theres possibility that somewhere $token isnt used as string and right now you broke some (poorly written but working) functionality. Maybe we need to check usages of this method?

File: models/classes/security/xsrf/Token.php [hunk 3]
```diff
@@ -43,49 +56,39 @@ public function __construct($data = [])
...
     {
         $this->token = $token;
     }
 
     /**
      * Set the microtime at which the token was created.
-     * @param float $timestamp
      */
-    public function setCreatedAt($timestamp)
+    public function setCreatedAt(float $timestamp): void
```
Comment (Karol-Stelmaczonek) [inline models/classes/security/xsrf/Token.php:67]: same as setValue()

File: models/classes/taskQueue/TaskLog/Entity/TaskLogEntity.php [hunk 4]
```diff
@@ -140,104 +144,69 @@ private static function parseDateTime(array $row, string $key, string $dateForma
...
+    public function getCreatedAt(): ?DateTime
     {
         return $this->createdAt;
     }
 
-    /**
-     * @return DateTime|null
-     */
-    public function getUpdatedAt()
+    public function getUpdatedAt(): DateTime
```
Comment (Karol-Stelmaczonek) [inline models/classes/taskQueue/TaskLog/Entity/TaskLogEntity.php:187]: It had DateTime|null in phpdoc not null is gone

**Review by Karol-Stelmaczonek — APPROVED** [review]
Need only to check those changed public methods and rest is ok.

**Pull request closed without merging**